tools/explorer/pkg/stellar/stellar.go

```diff
@@ -304,7 +307,7 @@ func (w *Wallet) GetBalance(address string, id schema.ID, asset Asset) (xdr.Int6
 	}
 
 	donors := make(map[string]struct{})
-	for len(txes.Embedded.Records) != 0 {
+	for len(txes.Embedded.Records) < stellarPageLimit {
```

Comment:
Comming to think of it, this check won't work if its put here. In here we need an unconditional for loop (otherwise we won't process txes from the last page if its not completely full), and the check needs to go near the end, just before we load more txes. if the check passes, we break out. Something like that should work I think